net-mgmt/xymon-client/src/etc/inc/plugins.inc.d/xymonclient.inc

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+    Copyright (c) 2024 Kumy
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+*/
+
+use Kumy\XymonClient\Settings;
+
+function xymonclient_enabled()
+{
+    $model = new Settings();
+    return (string)$model->enabled == '1';
+}
+
+function xymonclient_services()
+{
+    global $config;
+
+    $services = [];
+
+    $fqdn = sprintf('%s.%s',
+        $config['system']['hostname'],
+        $config['system']['domain'],
+    );
+
+    if (xymonclient_enabled()) {
+        $services[] = [
+            'description' => gettext('Xymon client'),
+            'configd' => [
+                'restart' => ['xymonclient restart'],
+                'start' => ['xymonclient start'],
+                'stop' => ['xymonclient stop'],
+            ],
+            'name' => 'xymonclient',
+            'pidfile' => "/usr/local/www/xymon/client/logs/clientlaunch.$fqdn.pid"
```

Comment:
```suggestion
            'pidfile' => "/var/run/xymonclient.pid"
```

Two reasons, when making the pid dynamic, restart behavior might get flaky and the usual directory is `/var/run` and not exposed via the webserver.

Reply:
@AdSchellevis 
This is not my choice. It's defined in file `/usr/local/etc/rc.d/xymon-client` from the package itself.

Excerpt:
```
#!/bin/sh

# PROVIDE: xymon_client
# REQUIRE: DAEMON
# KEYWORD: shutdown

. /etc/rc.subr

name=xymon_client
rcvar=xymon_client_enable

load_rc_config "$name"
: ${xymon_client_enable:=NO}
: ${xymon_client_user:=xymon}

pidfile="/usr/local/www/xymon/client/logs/clientlaunch.`hostname`.pid"
command=/usr/local/www/xymon/client/bin/xymonlaunch
command_args="--config=/usr/local/www/xymon/client/etc/clientlaunch.cfg --log=/usr/local/www/xymon/client/logs/clientlaunch.log --pidfile=${pidfile}"
start_precmd=xymon_precmd

[…]
```

The variable `pidfile` is not overridable as it uses `=` and not something like `: ${pidfile:=...}`

Reply:
we can change the port rc.d script to better handle an external pid. Eventually we need to do something about the FreeBSD port itself though.

Reply:
Ok, done see latest commit and PR opnsense/ports#206